scripts/elasticsearch_nodes.py

```diff
@@ -0,0 +1,123 @@
+"""
+This script is referenced by TerraForm as an external data source.
+Prints the current deployment's Elasticsearch node IDs as JSON to STDOUT. If
+the list of node IDs is less than the number of placeholders on the Cloudwatch
+dashboard template, the output is padded with 'blank' entries.
+Example:
+{"nodes": "tVsGH2BSXXYe9sMLEYRcIv,ydG9meZMAGkTnAlPgLljz8,blank,blank"}
+"""
+from itertools import (
+    repeat,
+)
+import json
+import logging
+import re
+import sys
+from time import (
+    sleep,
+)
+
+from botocore.exceptions import (
+    ClientError,
+)
+
+from azul import (
+    config,
+    require,
+)
+from azul.deployment import (
+    aws,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+
+log = logging.getLogger(__name__)
+
+# How many times to retry the request to AWS if an error is encountered.
+max_attempts = 10
+
+
+def main() -> int:
+    i = iter(range(max_attempts))
+    while True:
+        attempt = next(i, None)
+        if attempt is None:
+            log.debug('Error: Max attempts exhausted')
+            return 1
```

Comment:
You somehow changed the semantics. If you check the loop condition at the top of the loop body you can use a normal loop, potentially with an `else` after the loop.